checks/apps/sirius/sirius_check.py

```diff
@@ -0,0 +1,185 @@
+        if ('stress' in self.output_data['ground_state'] and
+            'stress' in self.reference_data['ground_state']):
+            return sn.sum(
+                sn.abs(self.output_data['ground_state']['stress'][i][j] -
+                       self.reference_data['ground_state']['stress'][i][j])
+                for i in [0, 1, 2] for j in [0, 1, 2]
+            )
+        else:
+            return sn.abs(0)
+
+    @deferrable
+    def forces_diff(self):
+        ''' Return the difference between obtained and reference atomic forces'''
+        if ('forces' in self.output_data['ground_state'] and
+            'forces' in self.reference_data['ground_state']):
+            na = self.output_data['ground_state']['num_atoms']
+            return sn.sum(
+                sn.abs(self.output_data['ground_state']['forces'][i][j] -
+                       self.reference_data['ground_state']['forces'][i][j])
+                for i in range(na) for j in [0, 1, 2]
+            )
+        else:
+            return sn.abs(0)
+
+    @sanity_function
+    def assert_success(self):
+        return sn.all([
+            sn.assert_found(r'converged after', self.stdout,
+                            msg="Calculation didn't converge"),
+            sn.assert_lt(self.energy_diff(), 1e-5,
+                         msg="Total energy is different"),
+            sn.assert_lt(self.stress_diff(), 1e-5,
+                         msg="Stress tensor is different"),
+            sn.assert_lt(self.forces_diff(), 1e-5,
+                         msg="Atomic forces are different")
+        ])
+
+
+@rfm.simple_test
+class SARUS_sirius_scf_check(sirius_scf_base_test,
+                             SarusExtraLauncherOptionsMixin,
+                             CudaVisibleDevicesAllMixin):
+    container_image = variable(str, type(None), value=None)
+    valid_systems = ['+sarus']
+    valid_prog_environs = ['builtin']
+    tags = {'parallel_k'}
+
+    @run_after('setup')
+    def setup_container_platform(self):
+        # if container_image was provided then initialize a container execution
+        if self.container_image is not None:
+            self.container_platform = 'Sarus'
+            self.container_platform.image = self.container_image
+            self.container_platform.with_mpi = False
+            self.container_platform.pull_image = False
+            self.container_platform.command = (
+                f'{self.executable} {" ".join(self.executable_opts)}'
+            )
+        else:
+            raise ConfigError('container_image is not set')
+
+    def __init__(self):
+        super().__init__(4)
+
+@rfm.simple_test
+class UENV_sirius_scf_check(sirius_scf_base_test,
+                            ExtraLauncherOptionsMixin,
+                            CudaVisibleDevicesAllMixin):
+    valid_systems = ['+uenv -amdgpu']
+    valid_prog_environs = ['+sirius +mpi +openmp']
+    tags = {'parallel_k'}
+    def __init__(self):
+        super().__init__(4)
```

Comment:
Same here, you can set the `num_tasks` and `executable_opts` at the child class level

Reply:
fixed